tests/DCPS/Compiler/char_literals/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+cmake_minimum_required(VERSION 3.8...3.27)
+project(opendds_compiler_char_literals CXX)
+enable_testing()
+
+find_package(OpenDDS REQUIRED)
+include(opendds_testing)
+
+set(target_prefix "${PROJECT_NAME}_")
+set(src "${CMAKE_CURRENT_SOURCE_DIR}")
+set(dst ${CMAKE_CURRENT_BINARY_DIR})
+set(opendds_libs
+  OpenDDS::Dcps
+  TAO::TAO
```

Comment:
`TAO::TAO` is already be linked through:

https://github.com/OpenDDS/OpenDDS/blob/18103482a923d95ecb1370cf235d992e9529c0a8/dds/CMakeLists.txt#L5-L11

and:

https://github.com/OpenDDS/OpenDDS/blob/18103482a923d95ecb1370cf235d992e9529c0a8/dds/CMakeLists.txt#L561

I assume you're trying to fix:

> /home/runner/work/OpenDDS/OpenDDS/OpenDDS/build/tests/DCPS/Compiler/char_literals/opendds_compiler_char_literals_exe: error while loading shared libraries: libTAO_AnyTypeCode.so.3.1.1: cannot open shared object file: No such file or directory

This is weird. I'm assuming the Windows cmake builds are having the same problem. I'm not sure why this is only an issue in this test and if it really should be looking for the library why it can't find it as the TAO `lib` that should be in `PATH`/`LD_LIBRARY_PATH`.

I'll try to look at this in detail some time tomorrow. It might be that `TAO::AnyTypeCode` has to be added to `dcps_lib_deps` in Dcps' CMakeLists.

Reply:
Yeah I was, I saw it was linked from the Dcps lib, but just checked to see if it would work, doing the same with TAO::AnyTypeCode now as a long shot here